src/meta.jl

```diff
@@ -1,30 +1,51 @@
 module Meta
 import JSON
 import Polymake: appname_module_dict, module_appname_dict
+import Polymake: pm_Rational
 
 struct UnparsablePolymakeFunction <: Exception
     msg::String
-    UnparsablePolymakeFunction(function_name) = new("Cannot parse function: $function_name")
+    UnparsablePolymakeFunction(function_name, json) = new("Cannot parse function: $function_name\n$json")
 end
 
-pm_name_qualified(app_name, func_name) = "$app_name::$func_name"
+############### fuctions used at runtime (imported to App modules)
 
-function pm_name_qualified(app_name, func_name, templates)
-    qname = pm_name_qualified(app_name, func_name)
-    templs = length(templates) > 0 ? "<$(join(templates, ","))>" : ""
-    return qname*templs
+function polymake_arguments(args...; kwargs...)
+    isempty(kwargs) && return Any[ convert.(PolymakeType, args)... ]
+    return Any[ convert.(PolymakeType, args); pm_perl_OptionSet(kwargs) ]
 end
 
+function get_docs(input::String; full::Bool=true, html::Bool=false)
+    pos = UInt(max(length(input)-1, 0))
+    return Polymake.shell_context_help(input, pos, full, html)
+end
+
+function pm_name_qualified(app_name, func_name, templates=String[])
+    isempty(templates) && return
+    return "$app_name::$func_name<$(join(templates, ","))>"
+end
+
+translate_type_to_pm_string(::Type{Bool}) = "bool"
+translate_type_to_pm_string(::Type{Int32}) = "int"
+translate_type_to_pm_string(::Type{<:AbstractFloat}) = "Float"
+translate_type_to_pm_string(::Type{<:Rational}) = "Rational"
+translate_type_to_pm_string(::Type{<:pm_Rational}) = "Rational"
+translate_type_to_pm_string(::Type{<:Integer}) = "Integer"
+translate_type_to_pm_string(::typeof(min)) = "Min"
```

Comment:
I think it would be better to define structs 'Min' and 'Max' instead of overloading the min and max functions.

Reply:
@alexej-jordan has already implemented those for TropicalNumbers;
It's just a matter of convenience for the user; 
we can provide both overloads for `Type{pm_Min}` and `typeof(min)` so that both
```
Tropical.Polytope{pm_Min, Rational}(...)
Tropical.Polytope{min, Rational}(...)
```
work (I don't have strong preference for any of them).

Reply:
btw. both constructors `Tropical.Polytope{min}(...)` and `Tropical.Polytope{min, pm_Rational}` are defined